src/ethereum_test_specs/base_static.py

```diff
@@ -0,0 +1,229 @@
+
+        This method should be implemented by the subclasses.
+
+        The function returned can be optionally decorated with the `@pytest.mark.parametrize`
+        decorator to parametrize the test with the number of sub test cases.
+
+        Example:
+        ```
+        @pytest.mark.parametrize("n", [1])
+        @pytest.mark.parametrize("m", [1, 2])
+        @pytest.mark.valid_from("Homestead")
+        def test_state_filler(
+            state_test: StateTestFiller,
+            fork: Fork,
+            pre: Alloc,
+            n: int,
+            m: int,
+        ):
+            \"\"\"Generate a test from a static state filler.\"\"\"
+            assert n == 1
+            assert m in [1, 2]
+            env = Environment(**self.env.model_dump())
+            sender = pre.fund_eoa()
+            tx = Transaction(
+                ty=0x0,
+                nonce=0,
+                to=Address(0x1000),
+                gas_limit=500000,
+                protected=False if fork in [Frontier, Homestead] else True,
+                data="",
+                sender=sender,
+            )
+            state_test(env=env, pre=pre, post={}, tx=tx)
+
+        return test_state_filler
+        ```
+
+        To aid the generation of the test, the function can be defined and then the decorator be
+        applied after defining the function:
+
+        ```
+        def test_state_filler(
+            state_test: StateTestFiller,
+            fork: Fork,
+            pre: Alloc,
+            n: int,
+            m: int,
+        ):
+            ...
+        test_state_filler = pytest.mark.parametrize("n", [1])(test_state_filler)
+        test_state_filler = pytest.mark.parametrize("m", [1, 2])(test_state_filler)
+        if self.valid_from:
+            test_state_filler = pytest.mark.valid_from(self.valid_from)(test_state_filler)
+        if self.valid_until:
+            test_state_filler = pytest.mark.valid_until(self.valid_until)(test_state_filler)
+        return test_state_filler
+        ```
+
+        The function can contain the following parameters on top of the spec type parameter
+        (`state_test` in the example above):
+        - `fork`: The fork for which the test is currently being filled.
+        - `pre`: The pre-state of the test.
+
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def remove_comments(data: Dict) -> Dict:
+        """Remove comments from a dictionary."""
+        result = {}
+        for k, v in data.items():
+            if isinstance(k, str) and k.startswith("//"):
+                continue
+            if isinstance(v, dict):
+                v = BaseStaticTest.remove_comments(v)
+            elif isinstance(v, list):
+                v = [BaseStaticTest.remove_comments(i) if isinstance(i, dict) else i for i in v]
+            result[k] = v
+        return result
+
+    @model_validator(mode="before")
+    @classmethod
+    def remove_comments_from_model(cls, data: Any) -> Any:
```

Comment:
Instead of discarding these, they should be added to `_info`'s `description` field. Could be a bit messy, as they could have come from any location in the YAML? We can create a follow-up issue?

Reply:
I'll open an issue because I don't have a good idea on how to do it now 👍